frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx

```diff
@@ -82,6 +84,7 @@ function mapStateToProps(state) {
   return {
     datasetEditorTab: getDatasetEditorTab(state),
     isMetadataDirty: isResultsMetadataDirty(state),
+    hasResultsMetadata: getResultsMetadata(state) !== null,
```

Comment:
Not sure if it will work in all cases with this `null` check. Maybe make it less strict, i.e. `!= null` (not `!== null`)?